pkg/service/node.go

```diff
@@ -506,7 +516,14 @@ func (n *NodeService) NodeExpandVolume(ctx context.Context, req *csi.NodeExpandV
 // NodeGetInfo returns the node ID
 func (n *NodeService) NodeGetInfo(context.Context, *csi.NodeGetInfoRequest) (*csi.NodeGetInfoResponse, error) {
 	// the nodeID is the VM's ID in kubevirt or VMI.spec.domain.firmware.uuid
-	return &csi.NodeGetInfoResponse{NodeId: n.nodeID}, nil
+	// If node ID resolution failed, return a placeholder to allow CSI registration.
+	// Volume operations will fail with a descriptive error in ControllerPublishVolume.
+	nodeID := n.nodeID
+	if nodeID == "" && n.nodeIDErr != nil {
+		// Use format "unresolved/<nodeName>" so controller can include the node name in error messages
+		nodeID = "unresolved/" + n.nodeName
+	}
+	return &csi.NodeGetInfoResponse{NodeId: nodeID}, nil
```

Comment:
I think we're over complicating things a bit; we can just exit on any errors to build a proper nodeID.
There is no point starting the services in that case IMO

Reply:
Good point.
The initial thought was to avoid the kubevirt-csi-node pods being in Crashloop due to this, which doesn't look nice, and the event on the pod that tries to mount the PVC will tell the users that the vm info could not be resolved.

I'll update the PR so the pods will fail instead.